src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php

```diff
@@ -13,13 +13,20 @@
 
 use Symfony\Component\Messenger\Envelope;
 use Symfony\Component\Messenger\Exception\MessageDecodingFailedException;
+use Symfony\Component\Messenger\Stamp\MessageDecodingFailedStamp;
 use Symfony\Component\Messenger\Stamp\NonSendableStampInterface;
 
 /**
  * @author Ryan Weaver<ryan@symfonycasts.com>
  */
 class PhpSerializer implements SerializerInterface
 {
+    /**
+     * @internal
+     */
+    public static bool $createClassNotFound = false;
+    private static bool $classNotFoundDetected = false;
```

Comment:
I think I would prefer to make theses properties as instance properties. 
It won't assure that if the developer change the used Serializer you will still have it so you can maybe also add an interface to ensure they exists and set their values only if the Serializer implements that interface ?

Reply:
This is not possible since **static** `handleUnserializeCallback()` method use it